lib/core/src/sockComm.cpp

```diff
@@ -489,7 +489,8 @@ sockOpenForInConn( rsComm_t *rsComm, int *portNum, char **addr, int proto ) {
                     std::strncpy(*addr, alias->data(), alias->size());
                 }
 
-                const std::chrono::seconds age{irods::get_hostname_cache_eviction_age()};
+                const auto cache_config = irods::get_advanced_setting<nlohmann::json>(irods::KW_CFG_HOSTNAME_CACHE);
+                const std::chrono::seconds age{cache_config.at(irods::KW_CFG_EVICTION_AGE_IN_SECONDS).get<int>()};
```

Comment:
I realize string allocations can throw and we are leaning on the schema validation, but I think we should handle `nlohmann::json` exceptions and/or `std::exception` around this.

Reply:
Agreed. The original code handled exceptions and this should do the same.

Reply:
Done.